src/sys/options.cxx

```diff
@@ -615,3 +672,33 @@ std::map<std::string, const Options *> Options::subsections() const {
   }
   return sections;
 }
+
+namespace {
+void toString(const Options& value, std::string& fout) {
+  std::string section_name = value.str();
+
+  if (not section_name.empty()) {
+    fout += fmt::format("[{}]\n", section_name);
+  }
+
+  for (const auto& child : value.getChildren()) {
+    if (child.second.isValue()) {
+      const auto value = bout::utils::variantToString(child.second.value);
+      // Convert empty strings to ""
+      const std::string as_str = value.empty() ? "\"\"" : value;
+      fout += fmt::format("{} = {}\n", child.first, as_str);
+    }
+  }
+
+  for (const auto& subsection : value.subsections()) {
+    fout += "\n";
+    toString(*subsection.second, fout);
+  }
+}
+} // namespace
+
+std::string toString(const Options& value) {
+  std::string ss;
+  toString(value, ss);
```

Comment:
Why the two-step thing here, and not just have `std::string toString(const Options& value)`? Is there another place where `void toString(const Options& value, std::string& fout)` is useful?

Reply:
Probably not, I copied the implementation from the python version of `BoutOptions` in `data.py` -- I had a vague memory that the two-step-ness was needed there, but this took so long to get working, I didn't want to spent more time refining it already. Will have another look when I come back to this!